addons/binding/org.openhab.binding.plugwise/src/main/java/org/openhab/binding/plugwise/handler/AbstractSleepingEndDeviceHandler.java

```diff
@@ -0,0 +1,98 @@
+
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.openhab.binding.plugwise.internal.protocol.AcknowledgementMessage;
+import org.openhab.binding.plugwise.internal.protocol.AnnounceAwakeRequestMessage;
+import org.openhab.binding.plugwise.internal.protocol.AnnounceAwakeRequestMessage.AwakeReason;
+import org.openhab.binding.plugwise.internal.protocol.BroadcastGroupSwitchResponseMessage;
+import org.openhab.binding.plugwise.internal.protocol.InformationResponseMessage;
+import org.openhab.binding.plugwise.internal.protocol.Message;
+
+/**
+ * The {@link AbstractPlugwiseThingHandler} handles common Plugwise sleeping end device (SED) channel updates and
+ * commands.
+ *
+ * @author Wouter Born - Initial contribution
+ */
+public abstract class AbstractSleepingEndDeviceHandler extends AbstractPlugwiseThingHandler {
+
+    private static final int SED_PROPERTIES_COUNT = 3;
+
+    public AbstractSleepingEndDeviceHandler(Thing thing) {
+        super(thing);
+    }
+
+    protected abstract int getWakeupDuration();
+
+    protected void handleAcknowledgement(AcknowledgementMessage message) {
+        updateStatusOnDetailChange();
+    }
+
+    protected void handleAnnounceAwakeRequest(AnnounceAwakeRequestMessage message) {
+        AwakeReason awakeReason = message.getAwakeReason();
+        if (awakeReason == AwakeReason.MAINTENANCE || awakeReason == AwakeReason.WAKEUP_BUTTON
+                || editProperties().size() < SED_PROPERTIES_COUNT) {
+            updateInformation();
+            if (isConfigurationPending() && !recentlySendConfigurationUpdate()) {
+                sendConfigurationUpdateCommands();
+            }
+        }
+    }
+
+    protected void handleBroadcastGroupSwitchResponseMessage(BroadcastGroupSwitchResponseMessage message) {
+        updateState(CHANNEL_TRIGGERED, message.getPowerState() ? OnOffType.ON : OnOffType.OFF);
+    }
+
+    protected void handleInformationResponse(InformationResponseMessage message) {
+        updateProperties(message);
+    }
+
+    @Override
+    public void handleReponseMessage(Message message) {
+
+        updateLastSeen();
+
+        switch (message.getType()) {
+            case ACKNOWLEDGEMENT_V1:
+            case ACKNOWLEDGEMENT_V2:
+                handleAcknowledgement((AcknowledgementMessage) message);
+                break;
+            case ANNOUNCE_AWAKE_REQUEST:
+                handleAnnounceAwakeRequest((AnnounceAwakeRequestMessage) message);
+                break;
+            case BROADCAST_GROUP_SWITCH_RESPONSE:
+                handleBroadcastGroupSwitchResponseMessage((BroadcastGroupSwitchResponseMessage) message);
+                break;
+            case DEVICE_INFORMATION_RESPONSE:
+                handleInformationResponse((InformationResponseMessage) message);
+                break;
+            default:
+                break;
```

Comment:
Eclipse shows a warning when I remove the default case of a switch. I've added trace logging that is useful for debugging and also makes the default case more useful.